### DISCUSSION TITLE
Drivetrain and HSS/generator dynamics

### DISCUSSION THREAD

**Discussion by eletroinf** [OpenFAST/openfast#1220] [Q&A] [answered]
Hello,

I'm modeling a two-mass representation of the rotor, drivetrain (LSS + HSS), and generator system.
I'm using openFAST v3.2.1 with Simulink interface.

I'm working to understand what is really modeled inside the openFAST. When using Simulink, I need to feed the Sfunction with the rotor speed. So, my question is about what parts of the drivetrain have dynamics modeled inside the openFAST?
As I'm feeding openFAST with HSS torque, from my generator model.
I can still enable the flags DrTrDOF  (Drivetrain rotational-flexibility DOF (flag)) and  GenDOF (Generator DOF (flag)). Do these flags have any effect in my scenario as they have with internal openFAST generator models? 
And also we have this:
GenIner    This is the moment of inertia of the high-speed portion of the drivetrain including the gearbox, HSS, and generator. If torsional flexibility of the drivetrain is enabled, the compliance is between the inertia of the rotor and this inertia. This value will be multiplied by the square of the gear ratio to map it to the low-speed reference frame. This value must not be negative. (kg·m2)
But it does not say anything about using the Simulink generator model.

So, in other words, my goal is to build the generator model in Simulink, and I need to know if HSS dynamics, including the generator inertia, are being simulated in openFAST if the drivetrain dynamics flags are enabled (and, in this case, I set my Simulink generator model inertia and damping inside openfast?); Or only the rotor dynamics are being simulated inside openFAST in this case.

Regards,

Ricardo

**Comment by jjonkman** [comment, marked as answer]
Dear @eletroinf,

I'm not sure I understand this point from your e-mail:

> When using Simulink, I need to feed the Sfunction with the rotor speed.

 Instead of this, I would say that the OpenFAST S-Function needs as input from Simulink the electrical torque applied to the generator (i.e., the generator torque, expressed about the high-speed shaft) and electrical power.  The OpenFAST model within the S-Function accounts for the structural dynamics of the drivetrain--including the rigid body rotation of the drivetrain (enabled through `GenDOF`),the drivetrain torsional flexibility (enabled through `DrTrDOF`), and associated generator and hub inertia, spring and damper--and the coupling of the drivetrain to the rotor and nacelle/support structure.  Simulink should calculate the electrical torque transmitted between the generator and the nacelle.

For more information on the drivetrain model implemented in the ElastoDyn module of OpenFAST, please see the following topic on our forum: https://forums.nrel.gov/t/resistant-moment-of-the-rotor-and-of-the-electric-generator/408.

Best regards,

**Comment by eletroinf** [comment]
Dear @jjonkman 

You got my point and clarified my doubt.
Thank you!
